Android build error for ffmpeg-kit-react-native

While building a dev client for android (on EAS) the gradle step fails with 
```
[stderr] FAILURE: Build failed with an exception.
[stderr] * What went wrong:
[stderr] Execution failed for task ':app:mergeDebugNativeLibs'.
[stderr] > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
[stderr]    > More than one file was found with OS independent path 'lib/x86/libc++_shared.so'.
 If you are using jniLibs and CMake IMPORTED targets, 
see https://developer.android.com/studio/preview/features#automatic_packaging_of_prebuilt_dependencies_used_by_cmake
```
Which I've seen before and fixed for myself with a patch-package while integrating react-native-mmkv in my managed expo app.  [I found mention of this issue on the old version of ffmpeg-kit-react-native which was react-native-ffmpeg.](https://github.com/tanersener/react-native-ffmpeg/issues/54#issuecomment-479032006) The fix made it into the [Tips](https://github.com/tanersener/ffmpeg-kit/wiki/Tips#2-depending-another-android-library-containing-libc_sharedso) for the new repo. I believe this is a necessary step for the config plugin to work properly for android and the other ffmpeg binaries (I'm trying to use full-gpl). In the meantime, I'll try to write a patch for the plugin that incorporates this fix. Btw thanks Evan for the plugin. It works great on iOS and was essential for supporting cross-platform video sharing in my expo app.

--- a/packages/ffmpeg-kit-react-native/build/withAndroidFFMPEGPackage.js
+++ b/packages/ffmpeg-kit-react-native/build/withAndroidFFMPEGPackage.js
@@ -4,7 +4,7 @@ exports.addPackageName = exports.withAndroidFFMPEGPackage = void 0;
 const config_plugins_1 = require("@expo/config-plugins");
 const generateCode_1 = require("@expo/config-plugins/build/utils/generateCode");
 const withAndroidFFMPEGPackage = (config, packageName) => {
-    return (0, config_plugins_1.withProjectBuildGradle)(config, (config) => {
+    config = (0, config_plugins_1.withProjectBuildGradle)(config, (config) => {
         if (config.modResults.language === "groovy") {
             config.modResults.contents = addPackageName(config.modResults.contents, packageName);
         }
@@ -13,6 +13,8 @@ const withAndroidFFMPEGPackage = (config, packageName) => {
         }
         return config;
     });
+    config = withLibCppSharedSo(config);
+    return config;
 };
 exports.withAndroidFFMPEGPackage = withAndroidFFMPEGPackage;
 function addPackageName(src, packageName) {
@@ -51,4 +53,16 @@ function appendContents({ src, newSrc, tag, comment, }) {
         };
     }
     return { contents: src, didClear: false, didMerge: false };
+}
+// This is required if you have several libraries which include libc++_shared.so as a dependency.
+// See https://github.com/tanersener/ffmpeg-kit/wiki/Tips#2-depending-another-android-library-containing-libc_sharedso
+function withLibCppSharedSo(config) {
+    return (0, config_plugins_1.withGradleProperties)(config, (config) => {
+        config.modResults.push({
+            type: "property",
+            key: "android.packagingOptions.pickFirsts",
+            value: "lib/x86/libc++_shared.so,lib/x86_64/libc++_shared.so,lib/armeabi-v7a/libc++_shared.so,lib/arm64-v8a/libc++_shared.so",
+        });
+        return config;
+    });
 }

--- a/packages/ffmpeg-kit-react-native/src/withAndroidFFMPEGPackage.ts
+++ b/packages/ffmpeg-kit-react-native/src/withAndroidFFMPEGPackage.ts
@@ -1,15 +1,20 @@
-import { ConfigPlugin, withProjectBuildGradle } from "@expo/config-plugins";
+import {
+  ConfigPlugin,
+  withGradleProperties,
+  withProjectBuildGradle,
+} from "@expo/config-plugins";
 import {
   createGeneratedHeaderComment,
   MergeResults,
   removeGeneratedContents,
 } from "@expo/config-plugins/build/utils/generateCode";
+import { ExpoConfig } from "@expo/config-types";
 
 export const withAndroidFFMPEGPackage: ConfigPlugin<string | undefined> = (
   config,
   packageName
 ) => {
-  return withProjectBuildGradle(config, (config) => {
+  config = withProjectBuildGradle(config, (config) => {
     if (config.modResults.language === "groovy") {
       config.modResults.contents = addPackageName(
         config.modResults.contents,
@@ -22,6 +27,10 @@ export const withAndroidFFMPEGPackage: ConfigPlugin<string | undefined> = (
     }
     return config;
   });
+
+  config = withLibCppSharedSo(config);
+
+  return config;
 };
 
 export function addPackageName(src: string, packageName?: string): string {
@@ -71,4 +80,18 @@ function appendContents({
     };
   }
   return { contents: src, didClear: false, didMerge: false };
+}
+
+// This is required if you have several libraries which include libc++_shared.so as a dependency.
+// See https://github.com/tanersener/ffmpeg-kit/wiki/Tips#2-depending-another-android-library-containing-libc_sharedso
+function withLibCppSharedSo(config: ExpoConfig): ExpoConfig {
+  return withGradleProperties(config, (config) => {
+    config.modResults.push({
+      type: "property",
+      key: "android.packagingOptions.pickFirsts",
+      value:
+        "lib/x86/libc++_shared.so,lib/x86_64/libc++_shared.so,lib/armeabi-v7a/libc++_shared.so,lib/arm64-v8a/libc++_shared.so",
+    });
+    return config;
+  });
 }

